torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -18,6 +18,24 @@ using torch::autograd::Node;
 using torch::autograd::validate_outputs;
 using torch::autograd::variable_list;
 
+namespace {
+
+static std::atomic<int> num_threads_blocked_in_backward = 0;
```

Comment:
Should this instead be a member variable in DistEngine, especially because getDebugInfo() is a member method?

Reply:
The guard is not a member of DistEngine and would need access to this variable and I'd like to keep the Guard definition in the cpp file.